akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -287,32 +292,33 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
   /** Special state that allows collecting some incoming data before dispatching it either as strict or streamed entity */
   case class CollectingIncomingData(
     headers:               ParsedHeadersFrame,
-    nextStateEmpty:        StreamState,
-    nextStateStream:       IncomingStreamBuffer => StreamState,
     correlationAttributes: Map[AttributeKey[_], _],
-    collectedData:         ByteString) extends StreamState {
-    override def handle(event: StreamFrameEvent): StreamState = event match {
-      case DataFrame(streamId, endStream, payload) =>
```

Comment:
Before, we handled only `DataFrame` but non of the other cases necessary as seen in `ReceivingData`.